### ISSUE TITLE
t-SNE plot improvements?

### ISSUE THREAD

**Issue by alanngnet** [alanngnet/CoverHunterMPS#8]
Ideas for improving the` _cluster_plot()` function I added to `src/eval_testset.py` :
- investigate whether PCA reduction would be valuable, or in which circumstances (maybe label count>N?) it's needed. Add it if so.
- add option to training script to save an epoch-specific cluster plot at each epoch that calls `eval_for_map_with_feat`
- connect `extract_csi_features` so _cluster_plot can have awareness of exactly which labels and which samples were excluded from the training data and mark those in the plot.
